pages/index.js

```diff
@@ -1,69 +1,93 @@
-import Head from 'next/head'
-import Image from 'next/image'
-import styles from '../styles/Home.module.css'
+import { useState, useEffect } from 'react';
+import { Stack, Button } from '@mui/material';
+import saveAs from 'file-saver';
+import SelectMenu from '../components/SelectMenu';
+import CustomTable from '../components/CustomTable';
+import loadCardsData from './api/data';
+import { FDDHeaderRow, LumpSumHeaderRow, menuItems } from '../helper';
+import Head from 'next/head';
+import styles from '../styles/Home.module.css';
 
 export default function Home() {
-  return (
-    <div className={styles.container}>
-      <Head>
-        <title>Create Next App</title>
-        <meta name="description" content="Generated by create next app" />
-        <link rel="icon" href="/favicon.ico" />
-      </Head>
+  const [cardType, setCardType] = useState('');
+  const [data, setData] = useState([]);
+  const [headerRow, setHeaderRow] = useState([]);
+
+  useEffect(() => {
+    if (cardType !== '') {
+      loadCardsData(cardType).then(setData);
+      cardType === 'FDD'
+        ? setHeaderRow(FDDHeaderRow)
+        : setHeaderRow(LumpSumHeaderRow);
+    }
+  }, [cardType]);
+
+  const handleChange = (event) => {
+    setCardType(event.target.value);
+  };
+
+  const generateCSVObject = () => {
+    const csvArray = [];
+    //Get an array of all attribute names (Column Headers) except rateCardHash & deleted//
+    const headers = headerRow?.reduce((accumulator, current) => {
+      if (current.key !== 'rateCardHash' && current.key !== 'deleted') {
+        accumulator.push(current.key);
+      }
+      return accumulator;
+    }, []);
+    csvArray.push(headers.join(','));
 
-      <main className={styles.main}>
-        <h1 className={styles.title}>
-          Welcome to <a href="https://nextjs.org">Next.js!</a>
-        </h1>
+    //Get an array of all the values corresponding to the above extracted headers//
+    for (const row of data) {
+      const values = headers.map((header) => row[header]);
+      csvArray.push(values.join(','));
+    }
 
-        <p className={styles.description}>
-          Get started by editing{' '}
-          <code className={styles.code}>pages/index.js</code>
-        </p>
+    return csvArray.join('\n');
+  };
 
-        <div className={styles.grid}>
-          <a href="https://nextjs.org/docs" className={styles.card}>
-            <h2>Documentation &rarr;</h2>
-            <p>Find in-depth information about Next.js features and API.</p>
-          </a>
+  const saveFile = (csvData) => {
+    const blob = new Blob([csvData], { type: 'text/csv;charset=utf-8;' });
```

Comment:
why blob and not new File()

Reply:
Just tried out the `new File()` method and it works perfectly too. I just happened to come across the blob method while saving a csv file so implemented that. No particular preference.